src/OSPSuite.R/Services/ConcurrentSimulationRunner.cs

```diff
@@ -13,6 +13,37 @@
 
 namespace OSPSuite.R.Services
 {
+   public class SimulationWithBatchOptions
+   {
+      public IModelCoreSimulation Simulation { get; set; }
+      public List<SimulationBatchRunValues> SimulationBatchRunValues { get; } = new List<SimulationBatchRunValues>();
+      public SimulationBatchOptions SimulationBatchOptions { get; set; }
+      public void AddSimulationBatchRunValues(SimulationBatchRunValues simulationBatchRunValues)
+      {
+         SimulationBatchRunValues.Add(simulationBatchRunValues);
+      }
+   }
+
+   class SimulationBatchRunOptions
+   {
+      public IModelCoreSimulation Simulation { get; set; }
+      public SimulationBatchRunValues SimulationBatchRunValues { get; set; }
+      public SimulationBatchOptions SimulationBatchOptions { get; set; }
+   }
+
+   public class ConcurrentSimulationResults
+   {
+      public ConcurrentSimulationResults(string simulationId, SimulationResults results, string additionalId = null)
+      {
+         SimulationId = simulationId;
+         SimulationResults = results;
+         AdditionalId = additionalId;
```

Comment:
I cannot understand why this is used from the name. Can you explain ?

Reply:
SimulationId <- Id of the simulation that was used for obtaining the results
SimulationResults <- Results after running the simulation
AdditionalId <- In case of Batch, the Id of the SimulationBatchRunValues

Reply:
Can it be "not" in the case of a batch?